File: TreatUs/Treat_Api/Models/Treat.cs
using System.Collections.Generic;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Treat_Api.Models
{
    public class Treat
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        public string DateTime { get; set; }

        public TreatTypes Type { get; set; }

        public string OtherTreatType { get; set; }

        public Category Category { get; set; }

        public List<Providers> Providers { get; set; }

        public List<Restaurants> Restaurants { get; set; }

        public List<TreatUsers> TreatUsers { get; set; }

        public TStatus Status { get; set; }

        public UserEmailStatus UserEmailStatus { get; set; }
    }

    public enum TStatus
    {
        Pending, InvitationSent, ProviderRejected, AllProviersAccepted
    }

    public enum TreatTypes
    {
        Birthday, FirstSalary, Car, Other
    }

    public enum Category
    {
        Breakfast, Lunch, Dinner, TeaParty,Drinks,Desserts,Other
    }

    public enum UserEmailStatus
    {
        Pending, AllEmaiSent
    }

}

File: TreatUs/Client/src/app/models/Treat.ts
import { GeneralInfo } from "./GeneralInfo";
import { AdminRestaurants } from "./AdminRestaurants";
import { TreatStatus } from "./TreatStatus";
import { TreatUser } from "./TreatUser";
import { TreatUsers } from '../models/TreatUsers';
import { from } from "rxjs";
import { SelectedProviders } from "./SelectedProviders";
import { TreatType } from "./TreatType";
import { Restaurants } from "./Restaurants";
import { Categories } from "./Categories";
import { UserEmailStatus} from "./UserEmailStatus";

export class Treat{
     id: string;
     dateTime:string;
     type:TreatType;
     otherTreatType:string;
     category:Categories;
     providers:SelectedProviders[];  
     status: TreatStatus;
     userEmailStatus: UserEmailStatus;
     restaurants:Restaurants[];
     treatUsers:TreatUsers[];

  //    constructor(treatId: string, status: number) {
  //     this.id = treatId;
  //     this.status = status
  // }
}